File: medium/us_date/us_date.php
<!DOCTYPE html>
<html>
<head>
	<title>Us date</title>
</head>
<body>
	<form  method="post">
		<label for="date">Link : </label>
		<input type="text"  name="date">

		<input type="submit" value="Action" >
	</form>
	<br>
	<?php

		function date_converted($date){

			$months = array('January'=> '1','February'=> '2',
				'March'=> '3',
				'April'=> '4',
				'May'=> '5',
				'June'=> '6',
				'July'=> '7',
				'August'=> '8',
				'September'=> '9',
				'October'=> '10',
				'November'=> '11',
				'December'=> '12');

			if ($date == NULL ){
				return("");
			}

			if (strpos($date,"/") !== false){
				$date_list = explode("/", $date);
				echo $date_list[1]."/".$date_list[0]."/".$date_list[2]."<br>";

			}else{
				$date_list = explode(" ", $date);
			
				echo $date_list[1][0]."/".$months[$date_list[0]]."/".$date_list[2];
			}
			
		}

	    $date = $_POST["date"];

	    date_converted($date);
        
	?>

</body>
</html>
		

File: medium/us_date/us_date.c
#include <stdio.h>
#include <string.h>
#define elif else if

int main ()
{
    char us[25]={};
    scanf("%[^\n]",us); getchar ();

    int d, m, y;
    if (us[1]=='/' || us[2]=='/')
    {
        sscanf (us, "%d/%d/%d", &m, &d, &y);
    }
    else
    {
        char month[20]={};
        sscanf (us, "%s %d, %d", month, &d, &y);

        if (strcmp (month, "January")==0) m=1;
        elif (strcmp (month, "February")==0) m=2;
        elif (strcmp (month, "March")==0) m=3;
        elif (strcmp (month, "April")==0) m=4;
        elif (strcmp (month, "May")==0) m=5;
        elif (strcmp (month, "June")==0) m=6;
        elif (strcmp (month, "July")==0) m=7;
        elif (strcmp (month, "August")==0) m=8;
        elif (strcmp (month, "September")==0) m=9;
        elif (strcmp (month, "October")==0) m=10;
        elif (strcmp (month, "November")==0) m=11;
        elif (strcmp (month, "December")==0) m=12;
    }
    printf ("%d/%d/%d", d, m, y);

    return 0;
}
